### ISSUE TITLE
Poor documentation

### ISSUE THREAD

**Issue by Bandd-k** [kolya95/cpp_made_2019#1]
There are o comments in the code and no readme. It is not possible to understand what is code doing at all :/

**Comment by kolya95** [comment]
Yes, it is a real problem in this project, could you make a pull request?
